email_tracking/app.rb

```diff
@@ -0,0 +1,54 @@
+require 'sinatra/base'
+require 'easypost'
+require 'sendgrid-ruby'
+require 'tilt/erb'
+require 'pry'
+require 'dotenv'
+
+class App < Sinatra::Base
+  set :show_exceptions, :after_handler
+
+  configure :development, :test do
+    Dotenv.load
+  end
+
+  configure do
+    EasyPost.api_key = ENV['EASYPOST_API_KEY']
+    set :sendgrid, SendGrid::API.new(api_key: ENV['SENDGRID_API_KEY'])
+  end
+
+ post '/easypost-webhook' do
+    request.body.rewind
+    request_string = request.body.read
+    response = JSON.parse(request_string)
```

Comment:
You don't need to do this. Just change this to `event = EasyPost::Event.receive(request.body.read)`. You shouldn't need to `rewind`. I prefer to get an error here saying that there is no content that safeguarding against someone already reading it.